base/show.jl

```diff
@@ -67,6 +67,9 @@ The following properties are in common use:
    can be avoided (e.g. `[Float16(0)]` can be shown as "Float16[0.0]" instead
    of "Float16[Float16(0.0)]" : while displaying the elements of the array, the `:typeinfo`
    property will be set to `Float16`).
+ - `:color`: Boolean specifying whether ANSI color/escape codes are supported/expected.
```

Comment:
Longer term, I think we may want to try to distinguish between policy ("wants color") and mechanism ("ansi", "xterm256", "html"). But this PR is already strictly better than what we have now, so I think we should merge this and then consider iterating (as needed).

Reply:
We can add new IO context attributes that Base understand in the future as long as the behavior of the existing ones remains the same.